src/Storage/ContentRequest/Listing.php

```diff
@@ -153,16 +153,18 @@ protected function runGroupSort($results)
             return $results;
         }
         $grouped = [];
+        $resultTaxOrders = [];
         foreach ($results as $result) {
             $taxGroup = null;
             foreach ($result->getTaxonomy() as $taxonomy) {
                 if ($taxonomy->getTaxonomytype() == $result->getTaxonomy()->getGroupingTaxonomy()) {
                     $taxGroup = $taxonomy->getSlug();
                     $taxOrder = $taxonomy->getSortorder();
+                    $resultTaxOrders[$result->getId()] = $taxOrder;
```

Comment:
Not sure if we can get varying contenttypes here, then this part would be troublesome for getting the correct sortorder.

i.e. `pages/1` and `entries/1` in the results